### ISSUE TITLE
GHC 8 compatibility

### ISSUE THREAD

**Issue by xich** [ekmett/exceptions#52]
Now that GHC 8 RC 1 is out, would it be possible to bump the bounds so this could build with GHC 8.0?

I think all that is needed is bumps to upper bounds of transformers (`< 0.6`) and template-haskell (`< 2.12`). I would submit a patch, but wasn't sure what to do about transformers-compat. I just removed it locally, but I realize that isn't an option for everyone.

**Comment by xich** [comment]
Sorry, just noticed #51

**Comment by xich** [comment]
Noticed a release today, so I went to give it a try. I think it won't work until you bump the bound on `transformers-compat`:

```
$ cabal install temporary-1.2.0.3
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: temporary-1.2.0.3 (user goal)
trying: exceptions-0.8.1 (dependency of temporary-1.2.0.3)
trying: mtl-2.2.1/installed-3AH... (dependency of exceptions-0.8.1)
trying: transformers-compat-0.4.0.4 (dependency of exceptions-0.8.1)
trying: transformers-compat-0.4.0.4:-two
rejecting: transformers-compat-0.4.0.4:-three (conflict: mtl =>
transformers==0.5.0.0/installed-0.5..., transformers-compat-0.4.0.4:three =>
transformers>=0.4.1 && <0.5)
rejecting: transformers-compat-0.4.0.4:+three (manual flag can only be changed
explicitly)
Dependency tree exhaustively searched.
```

**Comment by asr** [comment]
:+1:

**Comment by RyanGlScott** [comment]
Fixed with the release of [`exceptions-0.8.2.1`](http://hackage.haskell.org/package/exceptions-0.8.2.1).
